lib/include/hse_ikvdb/cndb.h

```diff
@@ -28,371 +30,135 @@ struct kvdb_rparams;
-
-/**
- * cndb_open() - Open a cndb
- * @ds:   dataset
- * @read_only: readonly flag
- * @cndb_entries: max number of entries in a CNDB's in memory representation.
- * @oid1: mlog id #1 of the cndb MDC
- * @oid2: mlog id #2 of the cndb MDC
- * @health: reference to the container kvdb's health struct
- * @cndb_out: cndb handle
- *
- * Implementation notes:
- *   - Cndb MDC processing will occur as part of cndb_open, rolling MDC
- *     transactions forward or backward as appropriate.
- *   - Diagnostic tools such as cn_metrics, and cn_kbdump will use
- *     cn secret sauce APIs to interact with cn trees w/o fully
- *     activating them.
- */
+cndb_destroy(struct mpool *mp, uint64_t oid1, uint64_t oid2);
+
 /* MTF_MOCK */
 merr_t
 cndb_open(
-    struct mpool *      ds,
-    bool                read_only,
-    atomic_ulong       *ikvdb_seqno,
-    size_t              cndb_entries,
-    u64                 oid1,
-    u64                 oid2,
-    struct kvdb_health *health,
+    struct mpool        *mp,
+    u64                  oid1,
+    u64                  oid2,
+    bool                 rdonly,
     struct kvdb_rparams *rp,
-    struct cndb **      cndb_out);
+    struct kvdb_health  *health,
+    struct cndb        **cndb_out);
 
-/**
- * cndb_cn_close() - Release a reference on a specific cn
- * @cndb:         cndb handle
- * @cnid:         cnid to release
- */
 /* MTF_MOCK */
 merr_t
-cndb_cn_close(struct cndb *cndb, u64 cnid);
-
-/**
- * cndb_getref() - gain a reference to cndb
- * @cndb:         cndb handle
- */
-/* MTF_MOCK */
-void
-cndb_getref(struct cndb *cndb);
+cndb_close(struct cndb *cndb);
 
-/**
- * cndb_putref() - release a reference to cndb
- * @cndb:         cndb handle
- */
-/* MTF_MOCK */
-void
-cndb_putref(struct cndb *cndb);
-
-/**
- * cndb_replay() - read and compact cndb MDC in memory
- *
- * Only function that reads the cndb mdc.
- * Called in the context of a kvdb open.
- *
- * @cndb:       cndb handle
- * @seqno:      highest in-use seqno
- * @ingestid:   ingestid of the latest successful ingest
- *      If no successul ingest is found, CNDB_INVAL_INGESTID is returned.
- */
 /* MTF_MOCK */
 merr_t
 cndb_replay(struct cndb *cndb, u64 *seqno, u64 *ingestid, u64 *txhorizon);
 
-typedef merr_t
-cn_init_callback(void *, struct kvset_meta *, u64);
-
-/**
- * cndb_cn_instantiate() - create/instantiate a cn tree when a KVS is opened.
- *
- * Does not read the cndb mdc to get KVS information, instead get information
- * from memory (cndb->cndb_tagv).
- * Iterate through C/M records corresponding to kvsets
- *
- * @cndb:         cndb handle
- * @cnid:         target cnid
- * @ctx:          caller's context, passed to cb()
- * @cb:           callback routine to be invoked for each matching C/M tuple
- *                create kvsets in the new cn tree of the KVS
- */
 /* MTF_MOCK */
 merr_t
-cndb_cn_instantiate(struct cndb *cndb, u64 cnid, void *ctx, cn_init_callback *cb);
+cndb_compact(struct cndb *cndb);
```

Comment:
Seems like `cndb_compact` should be a private API.

Reply:
Yes, but testing it is simpler this way